### PR TITLE
Document cron and delta timetables

### PR DESCRIPTION
**Pull request by AlexisBRENON**
Add a comparison between delta and cron data interval timetable to highlight differences when catchup is False.

Related: #32346 

---

### CODE REVIEW

File: docs/apache-airflow/authoring-and-scheduling/timetable.rst
Comment (eladkal) [inline docs/apache-airflow/authoring-and-scheduling/timetable.rst]: This is not about TimeTables specifically.. it's about scheduling in general.
Is this the right place to document this?

I think this is better to be explained in 
https://airflow.apache.org/docs/apache-airflow/stable/core-concepts/dag-run.html#catchup
and reference Timetable doc to the catchup doc

File: docs/apache-airflow/authoring-and-scheduling/timetable.rst
Comment (uranusjr) [inline docs/apache-airflow/authoring-and-scheduling/timetable.rst]: This title is not right since all timetables have a data interval. I would say something like _Differences between using a cron and timedelta to express data intervals_.

File: docs/apache-airflow/authoring-and-scheduling/timetable.rst
Comment (uranusjr) [inline docs/apache-airflow/authoring-and-scheduling/timetable.rst]: They are still different if the `start_time` does not align with the cron expression. Say you have `start_date=datetime(2023, 1, 1, 1)`, a daily cron runs from midnight to midnight (starting from 2nd Jan), while a `timedelta(days=1)` schedule would run everyday at 1am (starting from 1st Jan).

File: docs/apache-airflow/authoring-and-scheduling/timetable.rst
Comment (AlexisBRENON) [reply docs/apache-airflow/authoring-and-scheduling/timetable.rst]: That's the only two that are name `...DataIntervalTimetable`. But no problem, I'll update it.

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
Thanks @AlexisBRENON ! Good one. I will leave it to @uranusjr and @eladkal to get their second pass before we merge it!

File: docs/apache-airflow/authoring-and-scheduling/timetable.rst
Comment (pragnareddye) [inline docs/apache-airflow/authoring-and-scheduling/timetable.rst]:
```suggestion
The following table summarizes the DAG runs created (and the data interval that they cover). if you enable
```

File: docs/apache-airflow/authoring-and-scheduling/timetable.rst
Comment (pragnareddye) [inline docs/apache-airflow/authoring-and-scheduling/timetable.rst]:
```suggestion
This timetable is more focused on the data interval value and does not necessarily aligned execution dates with
```

File: docs/apache-airflow/authoring-and-scheduling/timetable.rst
Comment (uranusjr) [inline docs/apache-airflow/authoring-and-scheduling/timetable.rst]:
```suggestion
If you enable a DAG at 01:05 on February 1st, the following table summarizes the DAG runs created (and the
```

Is this the intended change?

Comment (AlexisBRENON) [reply docs/apache-airflow/authoring-and-scheduling/timetable.rst]: Excuse my french :sweat_smile: 
Fixed

**Review by uranusjr — APPROVED** [review]

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
:tada:

**Pull request merged**
